Assets/MixedRealityToolkit.Examples/Demos/UX/Collections/Scripts/GridObjectLayoutControl.cs

```diff
@@ -0,0 +1,126 @@
+        [Tooltip("Optional text field to output the layout of control.")]
+        public TMPro.TextMeshPro text;
+
+        /// <summary>
+        /// Change the grid collection's layout to the next one in order.
+        /// </summary>
+        public void NextLayout()
+        {
+            GridObjectCollection.AnchorType cL = grid.Anchor;
+            var anchorTypeCount = Enum.GetNames(typeof(GridObjectCollection.AnchorType)).Length;
+            int n = (((int)cL) + 1) % anchorTypeCount;
+            grid.Anchor = (GridObjectCollection.AnchorType)n;
+            UpdateUI();
+        }
+
+        /// <summary>
+        /// Change the grid's layout to the previous one in order
+        /// </summary>
+        public void PreviousLayout()
+        {
+            GridObjectCollection.AnchorType cL = grid.Anchor;
+            var anchorTypeCount = Enum.GetNames(typeof(GridObjectCollection.AnchorType)).Length;
+            int n = (((int)cL) - 1) % anchorTypeCount;
+            grid.Anchor = (GridObjectCollection.AnchorType)n;
+            UpdateUI();
+        }
+
+        /// <summary>
+        /// Use this to run mock tests in grid object collection tests
+        /// and print the resulting positions of child objects to a file.
+        /// Used to get expected values for GridObjectCollectionTests.
+        /// </summary>
+        public void RunTest()
+        {
+            StartCoroutine(TestAnchors());
+        }
+
+
+        void Start()
+        {
+            UpdateUI();
+        }
+
+        void UpdateUI()
+        {
+            if (text != null)
+            {
+                text.text = "Anchor: " + grid.Anchor;
+            }
+            grid.UpdateCollection();
+        }
+
+        IEnumerator TestAnchors()
+        {
+            var go = new GameObject();
+            go.name = "grid";
+            var grid = go.AddComponent<GridObjectCollection>();
+            grid.Distance = 0.75f;
+            grid.CellWidth = 0.15f;
+            grid.CellHeight = 0.15f;
+
+            for (int i = 0; i < 3; i++)
+            {
+                var child = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                child.transform.parent = go.transform;
+                child.transform.localScale = Vector3.one * 0.1f;
+            }
+            grid.Layout = LayoutOrder.Horizontal;
+
+
+            var filename = "printgrid-" + DateTime.UtcNow.ToString("yyMMdd-HHmmss") + ".txt";
+            string path = Path.Combine(Application.persistentDataPath, filename);
+            StringBuilder sb = new StringBuilder();
+            sb.AppendLine("Vector3[] expectedPositions = new Vector3[] {");
+            foreach (GridObjectCollection.AnchorType et in Enum.GetValues(typeof(GridObjectCollection.AnchorType)))
+            {
+                grid.Anchor = et;
+                grid.UpdateCollection();
+                PrintGrid(grid, et.ToString(), path, sb);
+                yield return new WaitForSeconds(0.5f);
```

Comment:
Out of curiousity, what is the purpose of this .5 second wait for each anchor type? Is it waiting for the collection to have fully updated?

Reply:
No, it was so I could see the results :-) I'll make a comment.